src/lib.rs

```diff
@@ -308,14 +308,9 @@ where
                                 details.activity_id.clone(),
                             ));
```

Comment:
Should we record cancel requested flag in the `InflightActivityDetails` and avoid sending extra cancels on every heartbeat? This probably is minor, but I just don't see the reason to deliver cancellation multiple times.

Reply:
Oh, that's a really good idea... I'll add that.